presto-spark-base/src/main/java/com/facebook/presto/spark/execution/task/NativeExecutionTask.java

```diff
@@ -112,7 +112,10 @@ public NativeExecutionTask(
         if (!shuffleWriteInfo.isPresent()) {
             this.taskResultFetcher = Optional.of(new HttpNativeExecutionTaskResultFetcher(
                     updateScheduledExecutor,
+                    errorRetryScheduledExecutor,
                     this.workerClient,
+                    this.executor,
+                    queryManagerConfig.getRemoteTaskMaxErrorDuration(),
```

Comment:
I wonder if using the same value for get `/info` and `/results` is feasible, when you think about long-running tasks in production which do not produce any results for extended period of time and that the task fetcher starts as soon as the task starts and not when results are actually available. Would be good to ensure we are not breaking anything for those cases (not captured by unit tests)

Reply:
This should be Ok. No matter how long it takes for the task to complete, we should be able to keep polling for results successfully and getting 204 OK if results are not yet available.